test/utils/JsonYamlSchemaValidatorSpec.scala

```diff
@@ -0,0 +1,197 @@
+    }
+
+    "return a Left YamlParsingFailure when given malformed yaml" in {
+      val malformedYaml =
+        """
+      name: John
+      age: thirty  # Invalid type
+      hobbies: [reading, coding
+      """
+      val result = jsonYamlSchemaValidator.yamlToJson(malformedYaml)
+      result.isLeft shouldBe true
+    }
+  }
+
+  ".jsonToYaml()" should {
+
+    "convert valid json file to yaml" in {
+
+      val yamlStringFromFile = formatYaml(readFile("test/resources/sample.yaml"))
+      val jsonStringFromFile = formatJson(readFile("test/resources/sample.json"))
+
+      val result = jsonYamlSchemaValidator.jsonToYaml(jsonStringFromFile)
+      result shouldBe Right(yamlStringFromFile)
+    }
+
+    "return a Left parsing failure when empty string" in {
+      val result: Either[ParsingFailure, String] = jsonYamlSchemaValidator.jsonToYaml("")
+      result.isLeft shouldBe true
+    }
+
+    "return a Left failure when input string is invalid" in {
+      val result: Either[ParsingFailure, String] = jsonYamlSchemaValidator.jsonToYaml("{?")
+      result.isLeft shouldBe true
+    }
+  }
+
+  ".validateJson()" should {
+
+    "return true when validating a valid json payload against the correct json schema" in {
+
+      val result = jsonYamlSchemaValidator.validateJson(api1878SchemaJson, api1878DataJson)
+
+      result shouldBe Right(ReportSuccess)
+    }
+
+    "return Left ValidationFailure when validating a invalid json payload against the desired json schema" in {
+
+      val result = jsonYamlSchemaValidator.validateJson(api1878SchemaJson, invalidJsonData)
+      result shouldBe Left(ValidationFailure("instance type (object) does not match any allowed primitive type (allowed: [\"array\"])"))
+    }
+
+    "return Left ValidationFailure when validating a empty json payload against the desired json schema" in {
+
+      val invalidJsonData = "{}"
+
+      val result = jsonYamlSchemaValidator.validateJson(api1878SchemaJson, invalidJsonData)
+      result shouldBe Left(ValidationFailure("instance type (object) does not match any allowed primitive type (allowed: [\"array\"])"))
+    }
+
+    "return Left PayLoadParseFailure when validating a empty string against the desired json schema" in {
+
+      val invalidJsonData = ""
+
+      val result = jsonYamlSchemaValidator.validateJson(api1878SchemaJson, invalidJsonData)
+
+      result shouldBe Left(PayLoadParseFailure("no JSON Text to read from input\n at [Source: (StringReader); line: 1, column: 1]"))
+    }
+  }
+
+  ".validateJsonAgainstYamlSchema()" should {
+
+    "return Right(ReportSuccess) when validating a valid json payload against the correct yaml schema" in {
+
+      val result = jsonYamlSchemaValidator.validateJsonAgainstYamlSchema(api1878SchemaYaml, api1878DataJson)
+      result shouldBe Right(ReportSuccess)
+    }
+
+    "return Left ValidationFailure when json contains multiple invalid data values for date and status field" in {
```

Comment:
Added this test for multiple invalid data types, the .validateJson method returns, multiple errors and suggested data type formats